internal/git/git.go

```diff
@@ -18,45 +18,50 @@ package git
 import (
 	"bytes"
 	"context"
-	"crypto/x509"
 	"fmt"
 
 	"github.com/sigstore/gitsign/internal/fulcio"
 	"github.com/sigstore/gitsign/internal/signature"
 	"github.com/sigstore/gitsign/pkg/git"
 	"github.com/sigstore/gitsign/pkg/rekor"
-	"github.com/sigstore/rekor/pkg/generated/models"
 )
 
-func Sign(ctx context.Context, rekor rekor.Writer, ident *fulcio.Identity, data []byte, opts signature.SignOptions) ([]byte, *x509.Certificate, *models.LogEntryAnon, error) {
-	sig, cert, err := signature.Sign(ident, data, opts)
+func Sign(ctx context.Context, rekor rekor.Writer, ident *fulcio.Identity, data []byte, opts signature.SignOptions) (*signature.SignResponse, error) {
```

Comment:
Would it be better to have a separate `SignOffline`? The fact that most of this method is behind an early return that we expect to be hit most of the time is a little bit off-putting.

Reply:
Maybe factoring out all the Rekor stuff into a helper function would be better.

Reply:
Done.